File: program/JavaStudy/Shape/Circle.java
package Shape;

public class Circle {
	public double R;
	public double Area;
	public double Perimeter;
	public Circle(double r){
		this.R = r;
	}
	
	public void ShowArea(){
		this.Area = Math.PI * this.R * this.R;
		System.out.println("面积：" + this.Area);
	}
	public void ShowPerimeter(){
		this.Perimeter = 2 * Math.PI * R;
		System.out.println("周长：" + this.Perimeter);
	}
}

File: program/WebGIS从基础到开发实践/Chapter02/Sample2-4/js/com/WebGIS2/Graphics/Circle.js
define(["dojo/_base/declare", "./Shape"], function(declare, Shape) {	
	return declare( // 类名省略
		Shape, // 父类
		{
			radius: 0,

        		constructor: function(radius) {
            		this.radius = radius || this.radius;
        		},

        		setRadius: function(radius) {
            		this.radius = radius;
        		},

        		area: function() {
            		return Math.PI * this.radius * this.radius;
        		}
		}
	);
});